packages/ag-charts-community/src/scene/layer.ts

```diff
@@ -125,16 +105,95 @@ export class Layer extends Group {
             ctx.restore();
         }
 
+        ctx.restore();
+        ctx.verifyDepthZero?.(); // Check for save/restore depth of zero!
+
+        return true;
+    }
+
+    private renderNext(renderCtx: RenderContext, forceRender: boolean | 'dirtyTransform') {
+        if (!this.getVisibility()) {
+            return true;
+        } else if (!this.layer) {
+            super.render(renderCtx);
+            return false;
+        }
+
+        const { isDirty, isChildDirty, isChildLayerDirty } = this.isDirty(renderCtx);
+
+        if (forceRender) {
+            this.layer.clear();
+        }
+
+        if (isDirty || isChildDirty || isChildLayerDirty || forceRender) {
+            const layerRenderCtx: RenderContext = {
+                ...renderCtx,
+                ctx: this.layer.context,
+                forceRender,
+            };
+
+            super.render(layerRenderCtx, false);
+
+            this.layer.context.verifyDepthZero?.(); // Check for save/restore depth of zero!
+        } else {
+            this.debugSkip(renderCtx);
+            this.markClean({ recursive: false });
+        }
+
+        const { ctx } = renderCtx;
+        ctx.save();
+        ctx.resetTransform();
+        ctx.globalAlpha = this.getComputedOpacity();
+        this.layer.drawImage(ctx);
+        ctx.restore();
+
+        return true;
+    }
+
+    override render(renderCtx: RenderContext) {
+        const { opts: { name, mode } = {}, _debug: debug } = this;
+        const { isDirty, isChildDirty, isChildLayerDirty } = this.isDirty(renderCtx);
+
+        let { forceRender } = renderCtx;
+        const { stats } = renderCtx;
+
+        // If bounding-box of a layer changes, force re-render.
+        const currentBBox = this.getBBox();
+        if (!this.lastBBox?.equals(currentBBox)) {
+            forceRender = 'dirtyTransform';
+            this.lastBBox = currentBBox;
+        }
+
+        if (!isDirty && !isChildDirty && !isChildLayerDirty && !forceRender) {
+            this.debugSkip(renderCtx);
+            this.markClean({ recursive: false });
+            return; // Nothing to do.
+        }
+
+        if (forceRender !== 'dirtyTransform') {
+            forceRender = isChildDirty || this.dirtyZIndex;
+        }
+
+        if (this.dirtyZIndex) {
+            this.sortChildren(Group.compareChildren);
+        }
```

Comment:
Side-note: We should review the overlap with `Group.render()`, as much of the logic here is repeated there.